server/endpoint/health/searcher/endpoint_test.go

```diff
@@ -0,0 +1,149 @@
+			expectedHealth: "green",
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var err error
+
+			// Mock k8s api server handling `kubectl get aws/azure/kvmconfig <clusterID>`.
+			k8sAPIMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json; charset=utf-8")
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte(tc.k8sAPIResponse))
+			}))
+			defer k8sAPIMockServer.Close()
+
+			var restConfig *rest.Config
+			{
+				c := k8srestconfig.Config{
+					Logger: microloggertest.New(),
+
+					Address:   k8sAPIMockServer.URL,
+					InCluster: false,
+				}
+
+				restConfig, err = k8srestconfig.New(c)
+				if err != nil {
+					t.Fatal("expected", nil, "got", err)
+				}
+			}
+
+			k8sClient, err := kubernetes.NewForConfig(restConfig)
+			if err != nil {
+				t.Fatal("expected", nil, "got", err)
+			}
+
+			g8sClient, err := versioned.NewForConfig(restConfig)
+			if err != nil {
+				t.Fatal("expected", nil, "got", err)
+			}
+
+			f := flag.New()
+			v := viper.New()
+			v.Set(f.Service.Provider.Kind, "kvm")
```

Comment:
It feels to me like `"kvm"` should be test input and we can test for multiple providers.

Reply:
Done.